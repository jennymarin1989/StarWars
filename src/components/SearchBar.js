import React, { Component } from 'react';

class SearchBar extends Component {
  constructor(props){
    super(props);
    this.state = {
      inputSearch: ''
    }
  }

  render(){
    return(
      <div>
        <h1>Search People from StarWars</h1>
        <input id="search" value={this.props.state}/>
      </div>  
    );
  }

}
export default SearchBar;