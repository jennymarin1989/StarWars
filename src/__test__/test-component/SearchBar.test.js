import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from '../../components/SearchBar';

describe ('SearchBar', () => {

  let search;

  beforeEach( () => {
    search = shallow(<SearchBar />);
  });

  describe ('renders correctly', () => {
    it('renders components correctly', () => {
      expect(search).toMatchSnapshot();
    });

    it('renders search bar correctly', () => {
      expect(search.find('#search')).toHaveLength(1);
    })
  });

  describe ('initializes state ', () => {
    it('initializes search state successfully', () => {
      expect(search.state('inputSearch')).toEqual('');
    })
  })
})