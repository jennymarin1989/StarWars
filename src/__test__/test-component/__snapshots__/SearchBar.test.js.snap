// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly renders components correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Search People from StarWars
</h1>,
        <input
          id="search"
          value={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search People from StarWars",
        },
        "ref": null,
        "rendered": "Search People from StarWars",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "search",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Search People from StarWars
</h1>,
          <input
            id="search"
            value={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search People from StarWars",
          },
          "ref": null,
          "rendered": "Search People from StarWars",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "search",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
